src/hooks/useChartDataToCsv.ts

```diff
@@ -66,22 +111,53 @@ const getMapCsv = (series: any) => {
   return [headers, dataRows];
 };
 
+const getIsIntervalData = (series: any) => {
+  return (
+    series.filter((x: { confidence: boolean }) => x?.confidence === true)
+      .length > 0
+  );
+};
+
 const getChartCsv = (series: any, category: string, chartType: any) => {
   const isCompactBarData = chartType === "compact bar";
-  const seriesNames = series.map((s: any) => s.name);
+  const isIntervalData = getIsIntervalData(series);
 
-  const headers = isCompactBarData
-    ? [category, seriesNames[0]]
-    : [category, ...seriesNames];
+  const seriesNames = series.map((s: any) => s.name);
 
   const verticalChart = series[0].orientation === "v";
   const uniqueXValues = getUniqueXValues(series, verticalChart);
 
   let dataRows: any;
   isCompactBarData
     ? (dataRows = getCompactBarChartDataRows(series, uniqueXValues))
+    : isIntervalData
+    ? (dataRows = getLineChartDataRows(series, uniqueXValues, verticalChart))
     : (dataRows = getDataRows(series, uniqueXValues, verticalChart));
 
+  let headers: any[] = [];
+  if (isCompactBarData) {
```

Comment:
This might be my Go side coming out and not a JS-y way of doing things, so ignore if so:
Given that nothing else happens in this function after each case, I would move the `return` into the cases. This would allow the next case to be its own 'if' rather than an 'else if' and then the final else can just be the last statement before the main return. Or rather, flip the logic so the biggest thing doesn't have to happen inside an if case:

So:

```suggestion
  if (isCompactBarData) {
   return [ [category, seriesNames[0]], dataRows];
 }
 
 if (!isIntervalData) {
   return [ [category, ...seriesNames], dataRows];
 }
   
// lots of stuff
return [headers, dataRows];

```
or something like that?

Reply:
In Go we call this keeping the '[line of sight'](https://medium.com/@matryer/line-of-sight-in-code-186dd7cdea88) so the main path of logic in a function is always on the primary line of sight, and then edgier cases and more detailed logic are whats happening in the indented line of site. Like I say, if its' anti-JS to do things this way thats fine, its just ingrained in my head as a readability thing but might not apply across languages well.

Reply:
Good suggestion, makes sense, I've made the changes suggested